utils/build.ps1

```diff
@@ -494,6 +534,40 @@ function Fetch-Dependencies {
     Invoke-Program -OutNull msiexec.exe /lvx! $BinaryCache\toolchains\$LogFile /qn /a $BinaryCache\toolchains\WixAttachedContainer\$_ ALLUSERS=0 TARGETDIR=$TARGETDIR
   }
 
+  $PythonAMD64URL = "https://www.nuget.org/api/v2/package/python/$PythonVersion"
+  $PythonAMD64Hash = "ac43b491e9488ac926ed31c5594f0c9409a21ecbaf99dc7a93f8c7b24cf85867"
+
+  $PythonARM64URL = "https://www.nuget.org/api/v2/package/pythonarm64/$PythonVersion"
+  $PythonARM64Hash = "429ada77e7f30e4bd8ff22953a1f35f98b2728e84c9b1d006712561785641f69"
+
+  function Download-Python($ArchName) {
+    if (-not (Test-Path $BinaryCache\Python$ArchName-$PythonVersion.zip)) {
+      Write-Output "Python ($ArchName) not found. Downloading from nuget.org ..."
+      New-Item -ItemType Directory -ErrorAction Ignore $BinaryCache | Out-Null
+      $PythonURL = (Get-Variable -Name "Python${ArchName}URL").Value
+      $PythonHash = (Get-Variable -Name "Python${ArchName}Hash").Value
+      if ($ToBatch) {
+        Write-Output "curl.exe -sL $PythonURL -o $BinaryCache\Python$ArchName-$PythonVersion.zip"
+      } else {
+        $WebClient.DownloadFile($PythonURL, "$BinaryCache\Python$ArchName-$PythonVersion.zip")
+        $SHA256 = Get-FileHash -Path "$BinaryCache\Python$ArchName-$PythonVersion.zip" -Algorithm SHA256
+        if ($SHA256.Hash -ne $PythonHash) {
+          throw "Python SHA256 mismatch ($($SHA256.Hash) vs $PythonHash)"
+        }
+      }
+    }
+    if (-not $ToBatch) {
+      # TODO(compnerd) stamp/validate that we need to re-extract
+      New-Item -ItemType Directory -ErrorAction Ignore $BinaryCache\Python$ArchName-$PythonVersion | Out-Null
+      Write-Output "Extracting Python ($ArchName) ..."
+      Expand-Archive -Path $BinaryCache\Python$ArchName-$PythonVersion.zip -Destination $BinaryCache\Python$ArchName-$PythonVersion -Force
+    }
+  }
+  Download-Python $HostArchName
+  if ($IsCrossCompiling) {
+    Download-Python $BuildArchName
+  }
```

Comment:
Rather than having a python specific download routine, having a generic helper to download is better IMO. I have put up https://github.com/apple/swift/pull/73058 to make that change (which also is something that the android cross-compilation will use).

Reply:
Rebased on top of #73058, done.